### DISCUSSION TITLE
[BE201] Node.js 連不到 MySQL  Error: Packets out of order. Got: 1 Expected: 0

### DISCUSSION THREAD

**Discussion by lucktanya33** [Lidemy/forum#184] [後端相關問題] [answered]
## 環境 
Mac BigSur
Xampp

## 問題
在跟著 [BE201] 後端中階：Express 與 Sequelize 裡面 Node.js 串接 MySQL 影片～2:12 之前跟著操作但連不到 MySQL，會出現  Error: Packets out of order. Got: 1 Expected: 0
錯誤畫面
![](https://i.imgur.com/3c6whc5.png)

## 我的程式碼與資料庫畫面
程式碼目前是跟著老師影片示範的做
![](https://i.imgur.com/AuSrrkw.png)
資料庫畫面
![](https://i.imgur.com/8bbkCcn.png)
資料庫是從 XAMPP -> go to application -> MySQL 進去的

3306 有開啟
![](https://i.imgur.com/3aco4nL.png)

## 查到的方法＆資料
* [Error: Packets out of order. Got: 1 Expected: 0](https://stackoverflow.com/questions/60614989/error-packets-out-of-order-got-1-expected-0)
* [protocol packets out of order](https://stackoverflow.com/questions/46140902/protocol-packets-out-of-order)

查到比較相關的資料不是那麼多，主要是以上兩個連結（或是我分辨不出來是相近的QQ 以上兩篇都有說要 在 my.conf 設定max_allowed_packet=500M，這邊也有跟著修改 my.conf 但還是出現一樣的錯誤訊息

大概卡了有幾天解決不了此問題，不曉得有沒有人有遇過同樣狀況~或是我在哪個環節有問題 求鞭打或指點！
謝謝看完的大家！

**Comment by aszx87410** [comment, marked as answer]
這幾篇也可以看一下
然後改完設定記得要重開 mysql 才會生效

https://github.com/mysqljs/mysql/issues/1363
https://github.com/mysqljs/mysql/issues/2451
https://blog.csdn.net/bossxu_/article/details/117655210

**Reply by lucktanya33** [reply]
謝謝老師的提點，目前是依 [protocol packets out of order](https://stackoverflow.com/questions/46140902/protocol-packets-out-of-order)  這篇把MySQL 的設定檔 my.conf 裡面的 max_allowed_packet 改成 500M 就可以連到了。

這邊遇到一個比較特別的狀況是重開 MySQL 要把電腦裡面整個 XAMPP 應用程式都結束掉，再打開才有效。一開始我只有在 XAMPP 裡面按stop 再 start ，儘管把conf 檔那邊改為 500M 了還是一樣是連不到的狀況。提供大家參考！
